### DISCUSSION TITLE
Scoped enumerations in CBOT

### DISCUSSION THREAD

**Discussion by melex750** [colobot/colobot#1515] [Ideas]
```c++
enum class Color : int
{
    blue  = 0x0000FF,
    green = 0x00FF00,
    red   = 0xFF0000,
};


Color c = Color::green;
```
```c++
// Enum class as member of class
public class TestClass
{
    enum class Enum
    {
        valueA, valueB, valueC,
    };
};


TestClass::Enum x = TestClass::Enum::valueC;
```

In case this feature is approved, I have them 95% implemented in my local repo.

**Comment by tomaszkax86** [comment]
I would drop `class` from `enum class` and just make it `enum`. C++ has `enum class` because it's an alternative to `enum`. CBOT doesn't have enums so there's no need to replicate awkward syntax.

I think I'd also drop size specification, I don't think CBOT needs that kind of thing but that's up for debate.

**Reply by melex750** [reply]
It's a good idea, I'll drop the word ```class``` and just call it ```enum```.  And I'll look into removing the ```: int``` syntax.

**Comment by kkard2** [comment]
If this gets implemented, it would be very nice to replace current integer constants for object categories with enums (probably not replace, but create overloads for functions such as `radar()` to have backwards script compatibility).
